components/FilterForm/index.js

```diff
@@ -1,12 +1,33 @@
 import styled from "styled-components";
 import React, { useState } from "react";
-
-export default function FilterForm({ plants, onAddPreference }) {
-  const [plantSize, setPlantSize] = useState("");
-  const [sunlightRequirement, setSunlightRequirement] = useState("");
-  const [waterNeeds, setWaterNeeds] = useState("");
-  const [optimalTemperature, setOptimalTemperature] = useState("");
-  const [petFriendly, setPetFriendly] = useState("");
+import { useRouter } from "next/router";
+
+export default function FilterForm({
+  plants,
+  onAddPreference,
+  initialFilterSettings,
+  initialFilterTitle = "",
```

Comment:
the change you made here seems to apply for all the useStates you use in this component.

Since you have a lot of props that are potentially undefined and need a fallback solution, it seems to make sense to render the whole FilterForm component only when the values are defined.

Reply:
Hey Dominik 😊 thanks for your tip!
I had tried that with if `(!plants || plants.length === 0) {return null; }` or with `if (!plants) {return null; }`
but always got an error messages like:  
23:57  Error: React Hook "useState" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  react-hooks/rules-of-hooks

Reply:
This is just because you make the return before you call the hook

```jsx
if(something){
  return
}

const [state, setState] = useState();
```

Just move the early return under all the useStates.